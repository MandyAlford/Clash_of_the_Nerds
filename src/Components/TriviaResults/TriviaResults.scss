.results-card {
  background-color: rgba(255, 255, 255, 0.9);
  padding: 30px;
  border: 5px solid #662324;
  align-items: center;
  height: 300px;
  width: 350px;
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  margin-bottom: 20px;
}
