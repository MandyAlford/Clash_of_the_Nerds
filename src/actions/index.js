export const addUser = (user) => ({
    type: 'ADD_USER',
    user
})

export const addTrivia = (trivia) => ({
    type: 'ADD_TRIVIA',
    trivia
})

export const addCorrectGuess = (guess) => ({
  type: 'ADD_CORRECT_GUESS',
  guess
})

export const addIncorrectGuess = (guess) => ({
  type: 'ADD_INCORRECT_GUESS',
  guess
})

export const resetAnswers = () => ({
  type: 'RESET_STORE_ANSWERS'
})

export const saveUsersTrivia = (trivia) => ({
  type: 'SAVE_TRIVIA',
  trivia
})
