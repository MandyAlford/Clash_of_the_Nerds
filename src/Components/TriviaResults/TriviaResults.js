import React from 'react';
import './TriviaResults.scss';
import { saveUsersTrivia } from '../../actions';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

export const TriviaResults = (props) => {
  const updateStoreAnswers = () => {
    props.saveUsersTrivia(props.result)
  }
  return (
    <div className='results-card'>
      <h2>{props.result.question}</h2>
      <h4>Your Answer: {props.result.userGuess}</h4>
      <h4>Correct Answer: {props.result.correct_answer}</h4>
      <button type='button' onClick={updateStoreAnswers}>Save Card</button>
    </div>
  )
}

TriviaResults.propTypes = {
  result: PropTypes.array
}

export const mapDispatchToProps = dispatch => ({
  saveUsersTrivia: (trivia) => { dispatch(saveUsersTrivia(trivia)) },
})

export default connect(null, mapDispatchToProps)(TriviaResults)
