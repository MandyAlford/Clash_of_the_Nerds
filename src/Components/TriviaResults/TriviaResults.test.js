import React from 'react';
import { TriviaResults, mapDispatchToProps } from './TriviaResults';
import { shallow } from 'enzyme';
import { saveUsersTrivia } from '../../actions';
import { mockRandom } from 'jest-mock-random';


describe('TriviaResults', () => {
  let wrapper;

  beforeEach(() => {
    mockRandom([0.1]);
    wrapper = shallow(<TriviaResults
      result={[{
        category: 'Mythology',
        type: 'true/false',
        difficulty: 'easy',
        question: 'Loki dressed as a woman in norse mythology',
        correct_answer: 'true',
        incorrect_answers: ['false'],
        userGuess: 'true'
      },
      {
        category: 'Mythology',
        type: 'multiple',
        difficulty: 'medium',
        question: 'This Greek mythological figure is the god/goddess of battle strategy (among other things).',
        correct_answer: 'Athena',
        incorrect_answers: ['Ares','Artemis','Apollo'],
        userGuess: 'Athena'
      }]}
      />)
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should call dispatch with saveUsersTrivia action when saveUsersTrivia is invoked', () => {
    const mockDispatch = jest.fn()
    const mockResult = {
      category: 'Mythology',
      type: 'multiple',
      difficulty: 'medium',
      question: 'This Greek mythological figure is the god/goddess of battle strategy (among other things).',
      correct_answer: 'Athena',
      incorrect_answers: ['Ares','Artemis','Apollo'],
      userGuess: 'Athena'
    }
    const actionToDispatch = saveUsersTrivia(mockResult)
    const mappedProps = mapDispatchToProps(mockDispatch)
    mappedProps.saveUsersTrivia(mockResult)
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
  })

})
