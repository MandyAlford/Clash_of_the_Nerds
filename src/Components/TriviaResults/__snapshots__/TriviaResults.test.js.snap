// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TriviaResults should match snapshot 1`] = `
<div
  className="results-card"
>
  <h2 />
  <h4>
    Your Answer: 
  </h4>
  <h4>
    Correct Answer: 
  </h4>
  <button
    className="results-btn"
    onClick={[Function]}
    type="button"
  >
    Save Card
  </button>
</div>
`;
